### PR TITLE
chore: Resolve issues

### LINKED ISSUE #38: Change emoji reaction for error messages
# Expected Behavior

jack should react with a different emoji when an error occurs.

# Current Behavior

Jack reacts to all messages with the same emoji(CUSTOM JACK EMOJI).

## Steps to Reproduce

1. Join SRMKZILLA's discord server: https://community.srmkzilla.net
2. Write gibberish so that the bot throws an error.

When no error occurs:

image

When an error occurs:

image

### PR DESCRIPTION
**Pull request by alvinbengeorge**
# Description
This update contains mainly two changes:
- [x] Upgrade MongoDB to `4.1.2`
- [x] ❌ Emoji for command errors
image

Fixes #38 #39 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] Tested using Discord

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/utils/nodecache.ts [hunk 1]
```diff
@@ -17,15 +17,15 @@ export const initCache = async (): Promise<nodeCache> => {
 export const refreshKeys = async () => {
   try {
     const db = (await getDbClient()).db();
-    const events = await db.collection("events").find<eventSchema>();
+    const events = await db.collection("events").find<eventSchema>({});
```
Comment (HarshPatel5940) [inline src/utils/nodecache.ts:20]: Hmm.... Looks good.

**Review by HarshPatel5940 — APPROVED** [review]
LGTM 🚀

File: src/helper/polls.ts [hunk 2]
```diff
@@ -97,8 +97,8 @@ export const createPoll = async (
 
 const addPoll = async (incomingMessage: Message, poll: pollSchema) => {
   const db = await (await getDbClient()).db().collection("polls");
-  const { insertedCount } = await db.insertOne(poll);
-  if (insertedCount <= 0) throw "MongoDB Poll Insert Error";
+  const { acknowledged } = await db.insertOne(poll);
+  if (!acknowledged) throw "MongoDB Poll Insert Error";
```
[the remark this answers is not in the document]
Comment (alvinbengeorge) [reply src/helper/polls.ts:101]: Had no choice lol

File: src/utils/nodecache.ts (hunk 1 above)
Comment (alvinbengeorge) [reply src/utils/nodecache.ts:20]: Seems to be a new required field

File: index.ts [hunk 3]
```diff
@@ -64,21 +64,22 @@ async function createServer() {
             ]),
           },
         };
+        let isError: boolean = false;
```
Comment (shawshankkumar) [inline index.ts:67]: You don't have to declare a type. If you just put false, it is automatically infered. Inference is better than straight up decl

File: src/utils/database.ts [hunk 4]
```diff
@@ -3,11 +3,7 @@ let dbClient: MongoClient;
 let eventDbClient: MongoClient;
 export async function initDbClient() {
   try {
-    dbClient = await MongoClient.connect(process.env.DATABASE_URI || "", {
-      useNewUrlParser: true,
-      useUnifiedTopology: true,
-      ignoreUndefined: true,
-    });
+    dbClient = await MongoClient.connect(process.env.DATABASE_URI || "");
```
Comment (shawshankkumar) [inline src/utils/database.ts:6]: Throw an error and do lrocess.exit if db not connected or db string not found

**Review by shawshankkumar — CHANGES_REQUESTED** [review]

**Comment by HarshPatel5940** [comment]
Looks good Overall, just had only one question what if someone updated your user permissions of that cluster and you are not able to update the data as you lack perms.

Will the usage `!acknowledge` still work as intended when compared to using `modifiedCount`?

>  Please verify this usecase.

**Comment by alvinbengeorge** [comment]
> Looks good Overall, just had only one question what if someone updated your user permissions of that cluster and you are not able to update the data as you lack perms.
> 
> Will the usage `!acknowledge` still work as intended when compared to using `modifiedCount`?
> 
> >  Please verify this usecase.

Any Idea how to check if it works, you know what, I'll do modified count but I didn't know how to use that or where to get that, I'll figure that out

**Comment by HarshPatel5940** [comment]
> Any Idea how to check if it works, you know what, I'll do modified count but I didn't know how to use that or where to get that, I'll figure that out
> 
> I'm pretty sure the purpose of that count was to check if it works, and I'm sure that acknowledged attribute will do it

Either check docs or just console log an mongo request. You will be able to find `modifiedCount`.

Also, if the *acknowledge* attribute works then we can go with it, not a problem. 🫡

File: src/utils/database.ts (hunk 4 above)
Comment (alvinbengeorge) [reply src/utils/database.ts:6]: image
It doesn't exit when I give process.exit()

**Comment by shawshankkumar** [comment]
@alvinbengeorge @HarshPatel5940 test this and lemme know

cc: @sm-sami

File: src/utils/database.ts (hunk 4 above)
Comment (alvinbengeorge) [reply src/utils/database.ts]: Completed

File: src/utils/database.ts [hunk 5]
```diff
@@ -24,18 +22,15 @@ export async function getDbClient() {
 
 export async function initEventDbClient() {
   try {
+    if (!process.env.EVENT_DATABASE_URI) throw new Error("No Event Database URI");
```
Comment (HarshPatel5940) [inline src/utils/database.ts:25]: do `process.exit(1)` here

File: src/utils/database.ts [hunk 6]
```diff
@@ -3,15 +3,13 @@ let dbClient: MongoClient;
...
-      ignoreUndefined: true,
-    });
+    if (!process.env.DATABASE_URI) throw new Error("No Database URI");
+    dbClient = await MongoClient.connect(process.env.DATABASE_URI);
     console.log("✔️   Connected to Database");
     return dbClient;
   } catch (error) {
-    throw error;
+    console.log("❌   Failed to connect to Database");
+    process.exit(1);
```
Comment (HarshPatel5940) [inline src/utils/database.ts:12]: revert this

**Review by HarshPatel5940 — CHANGES_REQUESTED** [review]

**Review by sm-sami — APPROVED** [review]
LGTM

**Pull request merged**